test/openjd/model/test_fuzz.py

```diff
@@ -0,0 +1,298 @@
+        {**base, "steps": [fuzz_step() for _ in range(random.randint(1, 3))]},
+        {**base, "steps": [{}]},
+        {**base, "parameterDefinitions": []},
+        {**base, "parameterDefinitions": [fuzz_job_parameter()]},
+        {**base, "jobEnvironments": [{}]},
+        {**base, random_string(1, 15): random_value()},
+        base,
+    ]
+    return random.choice(mutations)
+
+
+def fuzz_environment_template():
+    base = {
+        "specificationVersion": "environment-2023-09",
+        "environment": {
+            "name": "Env1",
+            "script": {"actions": {"onEnter": {"command": "echo"}}},
+        },
+    }
+
+    mutations = [
+        {},
+        {"specificationVersion": fuzz_spec_version()},
+        {"specificationVersion": "environment-2023-09"},
+        {"specificationVersion": "environment-2023-09", "environment": {}},
+        {**base, "specificationVersion": None},
+        {**base, "specificationVersion": "jobtemplate-2023-09"},
+        {**base, "environment": {}},
+        {**base, "environment": None},
+        {**base, "environment": {"name": ""}},
+        {**base, "environment": {"name": "Env", "script": {}}},
+        {**base, random_string(1, 15): random_value()},
+        base,
+    ]
+    return random.choice(mutations)
+
+
+def fuzz_yaml_json_string():
+    return random.choice(
+        [
+            "{}",
+            "[]",
+            "{",
+            "[",
+            '{"key": "value"}',
+            '{"key": }',
+            "key: value",
+            "- item1\n- item2",
+            "---\n...",
+            "\x00",
+            "",
+            "null",
+            "true",
+            "123",
+            '"string"',
+            random_string(0, 100),
+        ]
+    )
+
+
+def run_fuzzer(num_tests: int, seed: Optional[int], verbose: bool) -> bool:
+    if seed is not None:
+        random.seed(seed)
+    else:
+        seed = random.randint(0, 2**32 - 1)
+        random.seed(seed)
+
+    print(f"Fuzzing openjd-model with {num_tests} tests (seed={seed})...")
+    start_time = time.time()
+
+    crashes = []
+    successes = 0
+
+    for i in range(num_tests):
+        test_type = random.choice(["job", "env", "doc_json", "doc_yaml"])
+
+        try:
+            if test_type == "job":
+                template = fuzz_job_template()
+                decode_job_template(template=template)
+            elif test_type == "env":
+                template = fuzz_environment_template()
+                decode_environment_template(template=template)
+            elif test_type == "doc_json":
+                doc = fuzz_yaml_json_string()
+                document_string_to_object(document=doc, document_type=DocumentType.JSON)
+            else:
+                doc = fuzz_yaml_json_string()
+                document_string_to_object(document=doc, document_type=DocumentType.YAML)
+
+            successes += 1
+
+        except EXPECTED_EXCEPTIONS:
+            successes += 1
```

Comment:
There is one concern here. We expect some of these inputs to raise exceptions (e.g. `DecodeValidationError`) but others to not raise exceptions. But we do not have logic that to know which templates we expect to raise such errors.

So there is a chance that we are ignoring errors in our job/environment template decoder or allowing bad inputs through.

Is that something we are okay with? Are the goals of the test to only identify unhandled exceptions hidden in the decoder?